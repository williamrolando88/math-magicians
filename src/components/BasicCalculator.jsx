import React from 'react';
import Calculator from './Calculator';

const BasicCalculator = () => (
  <main className="flex items-center justify-between w-3/5">
    <h2 className="text-2xl">Give me something to calculate!</h2>
    <div>
      <Calculator />
    </div>
  </main>
);

export default BasicCalculator;
