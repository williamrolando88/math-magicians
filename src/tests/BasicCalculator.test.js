import React from 'react';
import renderer from 'react-test-renderer';
import BasicCalculator from '../components/BasicCalculator';

test('should render BasicCalculator', () => {
  expect(renderer.create(<BasicCalculator />)).toMatchSnapshot();
});
