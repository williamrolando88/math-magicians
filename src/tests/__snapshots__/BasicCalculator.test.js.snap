// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BasicCalculator 1`] = `
<main
  className="flex items-center justify-between w-3/5"
>
  <h2
    className="text-2xl"
  >
    Give me something to calculate!
  </h2>
  <div>
    <div
      className="grid grid-cols-4 w-max mx-auto mt-20"
    >
      <p
        className="col-span-4 bg-gray-500 text-white py-4 text-right px-2"
      >
        0
      </p>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-orange-500 hover:bg-orange-400 active:bg-orange-900"
        name="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-orange-500 hover:bg-orange-400 active:bg-orange-900"
        name="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-orange-500 hover:bg-orange-400 active:bg-orange-900"
        name="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-orange-500 hover:bg-orange-400 active:bg-orange-900"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-slate-50 hover:bg-slate-200 active:bg-slate-500"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="py-4 px-6 border active:text-white bg-orange-500 hover:bg-orange-400 active:bg-orange-900"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</main>
`;
